### DISCUSSION TITLE
How can let external tester to test firmware that is under embargo?

### DISCUSSION THREAD

**Discussion by sam412081go** [fwupd/fwupd#7308] [Q&A] [answered]
I have uploaded the firmware and moved it to the embargo stage. The external tester is not part of our LVFS vendor group, so they cannot access and download the vendor-embargo.conf file and use it for testing (e.g., fwupdmgr refresh && fwupdmgr update).

Or do I need to move the firmware to the testing stage to allow external testers to perform testing?

Thanks!

**Comment by superm1** [comment]
Does the external tester have a business relationship with your company? If so, you can express that relationship in LVFS and then when you upload firmware you can mark people part of that relationship as having access.

Or yes alternatively you can bring it to public testing.

**Reply by hughsie** [reply]
Or you can send them a link to the .cab file -- you can do `fwupdmgr install https://....` without auth I believe.

**Reply by sam412081go** [reply]
I can test the embargoed firmware on my side using 'fwupdmgr refresh & fwupdmgr update' to update device.

But the external testing vendor is using Moblab to test firmware updates on Chromebook(ChromeOS). If I move the firmware to the 'testing' stage, will they be able to perform the update test? What else do I need to prepare for them to perform the testing?

Thank you!

**Reply by hughsie** [reply, marked as answer]
> If I move the firmware to the 'testing' stage, will they be able to perform the update test?

They'd have to `fwupdmgr enable-remote lvfs-testing` -- but before that -- does https://lvfs.readthedocs.io/en/latest/moblab.html help?

**Reply by sam412081go** [reply]
Got it, I will continue to check it with testing vendor. Thank you!

**Reply by sam412081go** [reply]
How can I move the firmware to the 'Testing" stage? I have already uploaded one success report and there are no 'problems' tags in my LVFS web, but the button of 'Testing' shows 'Permissoin denied'.
![image](https://github.com/fwupd/fwupd/assets/17995855/05102a41-8d41-4bc3-985d-9bfe41c745ac)
![image](https://github.com/fwupd/fwupd/assets/17995855/30446797-eb9e-4fa0-b7ec-87cbc4688753)

**Reply by hughsie** [reply]
I've assigned you ACLs now, can you try again?

**Reply by sam412081go** [reply]
Yes, it's working now, thank you!
